test/emulation.spec.js

```diff
@@ -99,7 +99,12 @@ module.exports.addTests = function({testRunner, expect, puppeteer}) {
 
   describe('Page.emulateMedia', function() {
     it('should be an alias for Page.emulateMediaType', async({page, server}) => {
-      expect(page.emulateMedia).toEqual(page.emulateMediaType);
+      /* they aren't referentially equivalent
+       * so instead we check the bodies are the same
+       */
+      const emulateMediaStr = page.emulateMedia.toString();
+      const emulateMediaTypeStr = page.emulateMediaType.toString();
+      expect(emulateMediaStr).toEqual(emulateMediaTypeStr);
```

Comment:
@mathiasbynens I don't really understand this one; they are both methods on `Page.prototype` so I would expect them to be equivalent - but maybe I'm missing something obvious?

Reply:
Did the TypeScriptification cause a change in the (now generated) code? They're definitely supposed to be aliases: https://github.com/puppeteer/puppeteer/blob/f43468452a4397d72d5f64ad75bf41f362b7d4df/src/Page.js#L1149-L1150

Reply:
Like, does TypeScript translate that into something unexpected somehow?

Reply:
Nope - TypeScript outputs that line exactly :/

Is it something to do with how the test framework provides `page` ? I'm not sure.

Reply:
This is to do with the `installAsyncHooks` code that creates new methods. Fixed by doing the legacy binding in `index.js`.

Reply:
As discussed offline. Thanks for this drive-by fix!